src/main/java/appeng/spatial/StorageCellWorldData.java

```diff
@@ -0,0 +1,103 @@
+package appeng.spatial;
+
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.nbt.INBT;
+import net.minecraft.nbt.ListNBT;
+import net.minecraft.nbt.NBTUtil;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.storage.WorldSavedData;
+import net.minecraftforge.common.util.Constants;
+
+/**
+ * Extra data attached to the storage cell world.
+ */
+public class StorageCellWorldData extends WorldSavedData {
+
+    /**
+     * ID of this data when it is attached to a world.
+     */
+    public static final String ID = "ae2_storage_cell_world";
+
+    // Used to allow forward compatibility
+    private static final int CURRENT_FORMAT = 2;
+
+    private static final String TAG_FORMAT = "format";
+
+    private static final String TAG_LOTS = "lots";
+
+    private static final String TAG_LOT_ID = "id";
+
+    private static final String TAG_LOT_SIZE = "size";
+
+    private static final String TAG_LOT_OWNER = "owner";
+
+    private final Int2ObjectOpenHashMap<StorageCellLot> lots = new Int2ObjectOpenHashMap<>();
+
+    public StorageCellWorldData() {
+        super(ID);
+    }
+
+    public StorageCellLot getLotById(int id) {
+        return lots.get(id);
+    }
+
+    public StorageCellLot allocateLot(BlockPos size, int owner) {
+
+        int nextId = 1;
+        for (int id : lots.keySet()) {
```

Comment:
Might be worth to have an additional counter, which is initialized when reading the list of lots.
But for a small amount of cell it should be fine.